specfile/rpm.py

```diff
@@ -236,6 +237,51 @@ def reinit(cls, level_threshold: MacroLevel = MacroLevel.RPMRC) -> None:
 
 
 class RPM:
+    @staticmethod
+    @contextlib.contextmanager
+    def make_dummy_sources(sources: List[str], sourcedir: Path) -> Iterator[List[Path]]:
+        """
+        Context manager for creating temporary dummy sources to enable a spec file
+        to be fully parsed by RPM.
+
+        Args:
+            sources: List of all sources (including patches) in a spec file.
+            sourcedir: Path to sources and patches.
+
+        Yields:
+            List of paths to each created dummy source.
+        """
+        # based on rpmFileIsCompressed() in rpmio/rpmfileutil.c in RPM source
+        SIGNATURES = [
+            (".bz2", b"BZh"),
+            (".zip", b"PK00"),
+            (".xz", b"\xfd\x37\x7a\x58\x5a\x00"),
+            (".zst", b"\x28\xb5\x2f"),
+            (".lz", b"LZIP"),
+            (".lrz", b"LRZI"),
+            (".gz", b"\x1f\x8b"),
+            (".7z", b"7z\xbc\xaf\x27\x1c"),
+        ]
+        MAGIC_LENGTH = 13
```

Comment:
What does magic length mean? Do I understand it correctly that it's the number of characters that rpm takes a look at while parsing? If so, a comment would be nice

Reply:
Yes, RPM reads the first 13 bytes of a file to do signature matching. I thought it was self-explanatory, but I'll add a comment.